run-queue

```diff
@@ -191,11 +195,15 @@ def main():
                     return 1
 
             if isinstance(cmd, list):
-                cmd_str = ' '.join(cmd)
+                cmd_str = shlex.join(cmd)
             else:
                 cmd_str = cmd
             logging.info("Consuming message with command: %s", cmd_str)
-            p = subprocess.Popen(cmd, shell=isinstance(cmd, str))
+            if opts.wrapper:
+                cmd = [opts.wrapper, queue_name, cmd_str]
```

Comment:
I think I'd prefer to get the separate command arguments rather than the `shlex.join`-ed blob here.

I'm not sure the queue name is super useful as a way to help choose which container to use, particularly if we expand to a situation where we want to have per-git-repository task container tags.  I see a couple of options:

 - have the scanner side determine which container to use and include that in the queue somehow.  This would have the benefit of being able to be run a single time when doing the scanning (since all tasks queued on a given PR would use the same tasks container).
 - some kind of intelligence in the wrapper script — but this seems hard.  We actually don't even really know what git repository we're dealing with since the command is an opaque blob.
 - make the command to run a bit more structured — something like a git repository plus the command to run inside of that repository or something.  Then the wrapper script would have an easier job of figuring out what to do... but it would also be less like a wrapper and more like a beefed up `tests-invoke`...

Let's talk at breakfast :)

Reply:
Yeah, I don't know what I was thinking -- I had the same thought this night. I want to expose the additional information via env, like `$TASK_QUEUE`. I do want to retain the option of having a staging deployment to test infra or container changes without breaking production.

We can look into what it takes to wrap the queue contents into JSON, between tests-scan and run-queue. Shouldn't be hard.

Reply:
Wait, it already is a JSON blob. We just need to add the repo and branch name there. I'll do that in a separate PR.

Reply:
See #5917

Reply:
Yeah, I don't really like passing JSON to the wrapper script that much. I suppose we can make sure `jq` is installed, but I think env variables are a bit simpler. `run-queue` can pass `$TASK_REPO` and `$TASK_BASE` as env vars. However, if you prefer json we can also do that (feed it via stdin or `argv[1]`)? No strong opinion.

Reply:
Discussed IRL: Let's go with argv and env vars.